core/platform/search/gae_search_services_test.py

```diff
@@ -584,6 +594,12 @@ def test_search_using_single_sort_expression(self):
         self.assertEqual(result[1].get('id'), 'doc1')
         self.assertEqual(result[2].get('id'), 'doc2')
 
+    def test_raise_error_when_sort_starts_with_invalid_character(self):
+        doc = {'id': 'doc1', 'k': 'abc def', 'rank': 3, 'language_code': 'en'}
+        gae_search_services.add_documents_to_index([doc], 'index')
+        with self.assertRaises(ValueError):
+            gae_search_services.search('k:abc', 'index', sort='k')
```

Comment:
Why is this invalid? Perhaps give some explanation.

Reply:
done

Reply:
Is there some string in the error message raised that can ensure that the error happens because of what you mention in the comment? If so assert the occurrence of the message. ValueError is too generic.

Reply:
Done

Reply:
I don't think the regex captures the full essence of the error. It sounds too general. Can you add the full error message on this thread?

Reply:
done